### PR TITLE
Add timestreaminfluxdb db instance Update

### PR DESCRIPTION
**Pull request by trevorbonas**
### Description

- Update has been implemented for the `aws_timestreaminfluxdb_db_instance` resource.

### Relations

N/A

### References

N/A

### Output from Acceptance Testing

N/A

### CODE REVIEW

File: internal/service/timestreaminfluxdb/db_instance.go [hunk 1]
```diff
@@ -0,0 +1,855 @@
...
+	}
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.TimestreamInfluxDB, create.ErrActionSetting, ResNameDbInstance, plan.ID.String(), err),
+			err.Error(),
+		)
+		return
+	}
+	// Setting computed attributes
+	plan.Status = flex.StringToFramework(ctx, (*string)(&readOut.Status))
```
Comment (trevorbonas) [inline internal/service/timestreaminfluxdb/db_instance.go]: Computed attributes are usually updated above, around line 580, after calling the update method. However, the status always needs to be updated after any update and when tags are updated `UpdateDbInstance` cannot be used. Updating computed attributes here covers all cases.

File: internal/service/timestreaminfluxdb/db_instance.go [hunk 2]
```diff
@@ -196,8 +110,7 @@ func (r *resourceDbInstance) Schema(ctx context.Context, req resource.SchemaRequ
 				},
 				Description: `The name of the initial InfluxDB bucket. All InfluxDB data is stored in a bucket. 
 					A bucket combines the concept of a database and a retention period (the duration of time 
-					that each data point persists). A bucket belongs to an organization. This argument is optional.
```
Comment (forestmvey) [inline internal/service/timestreaminfluxdb/db_instance.go:199]: Is there a reason to remove the default value description?

File: internal/service/timestreaminfluxdb/db_instance.go [hunk 3]
```diff
@@ -239,64 +151,51 @@ func (r *resourceDbInstance) Schema(ctx context.Context, req resource.SchemaRequ
...
-					Possible values: 
-					"SINGLE_AZ", 
-					"WITH_MULTIAZ_STANDBY". 
-					If not provided, defaults to "SINGLE_AZ"`,
+					with a Multi-AZ standby for high availability.`,
 			},
 			"endpoint": schema.StringAttribute{
-				Computed: true,
+				Computed:    true,
+				Description: `The endpoint used to connect to InfluxDB. The default InfluxDB port is 8086.`,
```
Comment (forestmvey) [inline internal/service/timestreaminfluxdb/db_instance.go:193]: port 8086 is the only port that is supported.

File: internal/service/timestreaminfluxdb/db_instance.go (hunk 2 above)
Comment (trevorbonas) [reply internal/service/timestreaminfluxdb/db_instance.go:199]: I want all descriptions to match the AWS SDK Go. Terraform should communicate default values on its own, though I will double check this.

File: internal/service/timestreaminfluxdb/db_instance.go (hunk 3 above)
Comment (trevorbonas) [reply internal/service/timestreaminfluxdb/db_instance.go:193]: This is the description from AWS SDK Go. Link: https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/timestreaminfluxdb@v1.0.5#CreateDbInstanceOutput

File: internal/service/timestreaminfluxdb/db_instance.go (hunk 2 above)
Comment (trevorbonas) [reply internal/service/timestreaminfluxdb/db_instance.go:199]: Terraform does not communicate defaults or optional arguments when deploying or planning deployments. The comment above `StringAttribute.Description` reads:
```
	// Description is used in various tooling, like the language server, to
	// give practitioners more information about what this attribute is,
	// what it's for, and how it should be used. It should be written as
	// plain text, with no special formatting.
```

In this case, `StringAttribute.Optional`, `StringAttribute.Required`, and `StringAttribute.Validators` provide the information that I had previously included in the description.

File: internal/service/timestreaminfluxdb/db_instance.go [hunk 4]
```diff
@@ -310,15 +209,26 @@ func (r *resourceDbInstance) Schema(ctx context.Context, req resource.SchemaRequ
...
 					),
 				},
+				Description: `The name that uniquely identifies the DB instance when interacting with the 
+					Amazon Timestream for InfluxDB API and CLI commands. This name will also be a 
+					prefix included in the endpoint. DB instance names must be unique per customer 
+					and per region.`,
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
 			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
 			"organization": schema.StringAttribute{
```
Comment (forestmvey) [inline internal/service/timestreaminfluxdb/db_instance.go:219]: When running `terraform apply` with the configuration you noted [here](https://github.com/Bit-Quill/terraform-provider-aws/pull/2#issuecomment-2134094671), I got the following error:


```
Error: Provider produced inconsistent result after apply

When applying changes to aws_timestreaminfluxdb_db_instance.test_db_instance, provider "provider[\"registry.terraform.io/hashicorp/aws\"]" produced an unexpected new value:
.organization: was null, but now cty.StringVal("organization").

This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

The instance still deployed but I'm not sure what caused this error.

Comment (forestmvey) [reply internal/service/timestreaminfluxdb/db_instance.go:219]: Item was resolved after recompile.

**Review by forestmvey — APPROVED** [review]
Looks good

**Pull request merged**